src/client/Microsoft.Identity.Client/Cache/CacheKeyFactory.cs

```diff
@@ -48,12 +50,17 @@ public static string GetExternalCacheKeyFromResponse(
                 return key;
             }
 
-            if (requestParameters.AppConfig.IsConfidentialClient ||
```

Comment:
I recommend that you update the "internal" partition key as well. No point in having internal and external keys be different.

Reply:
Updated